functions-src/donate/main.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	stripe "github.com/stripe/stripe-go"
+	"github.com/stripe/stripe/checkout/session"
+)
+
+const (
+	recurringSKU = "plan_FZzqRxfQtAx34O"
+)
+
+func main() {
+	stripe.Key = os.Getenv("STRIPE_SECRET_KEY")
+	lambda.Start(handler)
+}
+
+type donationRequest struct {
+	Count     int  // SKU units, which are a euro each
+	Recurring bool // monthly, or one time only?
+}
+
+func handler(ctx context.Context, request events.APIGatewayProxyRequest) (*events.APIGatewayProxyResponse, error) {
+	// We expect a request body. Unmarshal it.
+	var req donationRequest
+	if err := json.Unmarshal([]byte(request.Body), &req); err != nil {
+		return nil, fmt.Errorf("unmarshal request: %w", err)
+	}
+
+	// Trivial validation
+	if req.Count <= 0 || req.Count > 1000 {
```

Comment:
Afraid of the whales?

Reply:
We've always had this restriction from years back :) Initially I think it was a safety net to avoid internal confusion about the .01 currency units thing and avoiding charging someone a meeelion dollars by misteak